src/renderer/App.tsx

```diff
@@ -117,6 +75,59 @@ function App() {
     listenOnlineStatus();
   }, []);
 
+  useEffect(() => {
+    ipcRenderer.once("set node info", (_, node: NodeInfo) => {
+      console.log(`set node info renderer App.tsx ${node}`);
+      setNodeInfo(node);
+    });
+  }, [nodeInfo]);
+
+  if (!nodeInfo) return null;
+
+  const headlessUrl = `${nodeInfo.host}:${nodeInfo.graphqlPort}`;
+  const wsLink = new WebSocketLink({
+    uri: `ws://${headlessUrl}/graphql`,
+    options: {
+      reconnect: true,
+    },
+  });
+
+  const httpLink = createHttpLink({ uri: `http://${headlessUrl}/graphql` });
+
+  const apiLink = split(
+    // split based on operation type
+    ({ query }) => {
+      const definition = getMainDefinition(query);
+      return (
+        definition.kind === "OperationDefinition" &&
+        definition.operation === "subscription"
+      );
+    },
+    wsLink,
+    httpLink
+  );
+
+  const link = ApolloLink.from([new RetryLink(), apiLink]);
+
+  const client = new ApolloClient({
+    link: link,
+    cache: new InMemoryCache(),
+  });
```

Comment:
These changes will cause a creation of a new Apollo Client on every render, which is highly undesirable. I know we gotta get URL via IPC first but still we gotta find a better way.